manifests/charts/istio-cni/values.yaml

```diff
@@ -40,11 +40,6 @@ defaults:
     # Some k8s flavors (e.g. OpenShift) do not support the chain approach, set to false if this is the case
     chained: true
 
-    # Allow the istio-cni container to run in privileged mode, needed for some platforms (e.g. OpenShift) or features (repairPods)
-    # Note that even if this is false, the `istio-cni` container *requires* root privileges on the node to function,
-    # and setting this to false does not change that, nor will it run the container as non-root or make it "un-privileged".
-    privileged: true
```

Comment:
I think openshift needs privilege for all operations, not just repair. I am not sure why exactly (it _might_ be the UDS log thing)

Reply:
I wanna test this with openshift first - it should be fine with `securityContext`

It might need `privileged: true` in order to actually manually drop the `securityContext` privs down in the first place - in which case `privileged` should still be hardcoded.

I also wanna see if we can drop CAP_SYS_ADMIN entirely in favor of more granular capabilities, as it's quite overloaded.

Reply:
good point John. cc a few openshift folks to chime in @dgn @rcernich @jwendell

Reply:
There is platform value in the bae chart. Not sure if this could be expanded to the cni chart.

Reply:
For now, I am hardcoding `privileged` to `true` if either repair or ambient are enabled.

It is redundant with CAP_SYS_ADMIN, and having both won't hurt, if one is required.


In general, it's better to rely on the mechanism of `drop: ALL` via `securityContext` and then explicitly adding back the CAP_XXX perms we actually require. 

Anything that doesn't trust us to run as `privileged` shouldn't/won't trust us to drop-all-add-SYS_ADMIN, and vice versa.